### PR TITLE
Cvs 82368 no camera fix

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

File: demos/horizontal_text_detection/python/horizontal_text_detection.py [hunk 1]
```diff
@@ -175,18 +175,27 @@ def run(self):
...
-    # crop and resize if original image is too small or too big for the model
-    if frame.shape[0] > HEIGHT and frame.shape[1] > WIDTH:
-        frame = frame[0:HEIGHT, 0:WIDTH]
-    else:
-        res = min(frame.shape[0], frame.shape[1])
-        frame = frame[0:res, 0:res]
-        frame = cv2.resize(frame, (HEIGHT, WIDTH), interpolation=cv2.INTER_AREA)
+    if frame == None:
+        print(f"[WARNING] No Input frame")
+    if frame != None:
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply demos/horizontal_text_detection/python/horizontal_text_detection.py:191]: Or even better, if frame == None just return frame immedietely. This way we can get rid of one indention level.

File: demos/horizontal_text_detection/python/horizontal_text_detection.py [hunk 2]
```diff
@@ -195,38 +204,40 @@ def grab_frame(cap):
 last_display_time = time.time()
 app_start_time = time.time()
 
-while(True):
-    if not threads[i].is_initialized():
-        threads[i].set_input(grab_frame(cap))
-        i = (i + 1) % args.num_threads
-        continue
+if grab_frame(cap) == None:
+    print(f"[ERROR] Check camera input...")
```
Comment (dkalinowski) [inline demos/horizontal_text_detection/python/horizontal_text_detection.py:208]: Same here, why cant we just quit an app if grab_frame returns None? This way we can save one indention level

File: demos/horizontal_text_detection/python/horizontal_text_detection.py [hunk 3]
```diff
@@ -195,38 +204,40 @@ def grab_frame(cap):
...
+else:
+    while(True):
+        if not threads[i].is_initialized():
+            threads[i].set_input(grab_frame(cap))
```
Comment (mzegla) [inline demos/horizontal_text_detection/python/horizontal_text_detection.py:212]: What if grab_frame returns None here as well?

File: demos/horizontal_text_detection/python/horizontal_text_detection.py [hunk 4]
```diff
@@ -175,18 +175,27 @@ def run(self):
 for thread in threads:
     thread.start()
 
+def finish():
+    global force_exit
```
Comment (mzegla) [inline demos/horizontal_text_detection/python/horizontal_text_detection.py:179]: `force_exit` is defined outside of any function scope. 
Do we need `global` keyword? 

same applies to line **139**.

File: demos/horizontal_text_detection/python/horizontal_text_detection.py (hunk 3 above)
Comment (rasapala) [reply demos/horizontal_text_detection/python/horizontal_text_detection.py:212]: Fixed.

File: demos/horizontal_text_detection/python/horizontal_text_detection.py [hunk 5]
```diff
@@ -194,8 +208,13 @@ def grab_frame(cap):
 frames_processed = 0
 last_display_time = time.time()
 app_start_time = time.time()
+cam_working = True
 
-while(True):
+if grab_frame(cap) == None:
+    print(f"[ERROR] Check camera input...")
+    cam_working = False
```
Comment (dkalinowski) [inline demos/horizontal_text_detection/python/horizontal_text_detection.py:215]: Why can't we use `force_exit` variable here?

File: demos/horizontal_text_detection/python/horizontal_text_detection.py [hunk 6]
```diff
@@ -194,8 +208,13 @@ def grab_frame(cap):
...
+
+while(cam_working):
```
Comment (dkalinowski) [inline demos/horizontal_text_detection/python/horizontal_text_detection.py:217]: Here to could be `while not force_exit:`

File: demos/horizontal_text_detection/python/horizontal_text_detection.py [hunk 7]
```diff
@@ -106,6 +106,9 @@ def __init__(self, index):
         self.predict_durations = []
         self.input_ready_event = threading.Event()
         self.output_ready_event = threading.Event()
+    
+    def exit():
```
Comment (dkalinowski) [inline demos/horizontal_text_detection/python/horizontal_text_detection.py:110]: Are you overriding `threading.Thread.exit`? It is never used in this file

**Review by dkalinowski — APPROVED** [review]

**Review by mzegla — APPROVED** [review]

**Pull request merged**